signal/signal.go

```diff
@@ -147,3 +153,13 @@ func (c *Interceptor) RequestShutdown() {
 func (c *Interceptor) ShutdownChannel() <-chan struct{} {
 	return c.shutdownChannel
 }
+
+// GetExitCode fetches the exit code that has been set of the Interceptor
+func (c *Interceptor) GetExitCode() int {
+	return <-c.exitCode
+}
+
+func (c *Interceptor) setExitCode(exitCode int) {
+	c.exitCode = make(chan int, exitCode)
```

Comment:
Channel needs to be created when we create the `Interceptor` otherwise the get/set of the channel itself could race. We also always need to buffer by 1, otherwise send on the channel will block.

Reply:
I had actually did the same :sweat_smile:  , by creating the channel while creating an interceptor with a buffer of one, something similar to the below snippet:
```go
	channels := Interceptor{
		interruptChannel:       make(chan os.Signal, 1),
		shutdownChannel:        make(chan struct{}),
		shutdownRequestChannel: make(chan struct{}),
		quit:                   make(chan struct{}),
		exitCode:               make(chan int, 1),
	}
```
everything did work, but the `go-simple` in Travis CI error-ed out telling me that I should not do this and just create a chan without any buffers, hence i had force-pushed the same. Ig, I will just ensure we call a `c.setExitCode()` always whenever lnd exits from any condition.